.jenkinsci/docker-pull-or-build.groovy

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env groovy
+
+def remoteFilesDiffer(f1, f2) {
+  sh "curl -L -o /tmp/${env.GIT_COMMIT}/f1 --create-dirs ${f1}"
+  sh "curl -L -o /tmp/${env.GIT_COMMIT}/f2 ${f2}"
+  diffExitCode = sh(script: "diff -q /tmp/${env.GIT_COMMIT}/f1 /tmp/${env.GIT_COMMIT}/f2", returnStatus: true)
+  if (diffExitCode == 0) {
+    return false
+  }
+  return true
+}
+
+def dockerPullOrUpdate() {
+  def platform = sh(script: 'uname -m', returnStdout: true).trim()
+  def commit = sh(script: "echo ${BRANCH_NAME} | md5sum | cut -c 1-8", returnStdout: true).trim()
+  if (remoteFilesDiffer("https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_COMMIT}/docker/develop/${platform}/Dockerfile", "https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_PREVIOUS_COMMIT}/docker/develop/${platform}/Dockerfile")) {
```

Comment:
what if try to write it in a manner like:
```
if ( remoteFilesDiffer( "https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_COMMIT}/docker/develop/${platform}/Dockerfile",
                        "https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_PREVIOUS_COMMIT}/docker/develop/${platform}/Dockerfile")) {
```
which is more readable.

Reply:
Also, what if there is no previous commits? `GIT_PREVIOUS_COMMIT = null`, or `GIT_PREVIOUS_COMMIT = GIT_COMMIT`?
In case of `null` - there might be errors from `curl` which might fail the whole pipeline. 

Regarding `diff` - in this case it will return non-zero exit code thus new docker image will be built and then uploaded to the dockerhub.

Reply:
According to [Jenkins GIt plugin](https://wiki.jenkins.io/display/JENKINS/Git+Plugin#GitPlugin-Environmentvariables), `GIT_PREVIOUS_COMMIT` equals `GIT_COMMIT` in case it is the first commit to that branch